### PR TITLE
Fix conditional update

### PR DESCRIPTION
**Pull request by msgroi**

### CODE REVIEW

File: src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ConditionMapper.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                        + primaryExpression + ", or filterExpression=" + filterExpression)));
+    }
+
+    /*
+     * Finds the value in the right-hand side operand of an expression where the left-hand operator is a given field.
+     */
+    @VisibleForTesting
+    static Optional<String> findVirtualValuePlaceholder(String conditionExpression, String keyFieldName) {
+        String toFind = keyFieldName + " = ";
+        int start = conditionExpression.indexOf(toFind);
```
Comment (busjaeger) [inline src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ConditionMapper.java]: What happens if they don't use spaces before and after equals sign?

File: src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ConditionMapper.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        if (start == -1) {
```
Comment (msgroi) [inline src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ConditionMapper.java]: Jeremy: what happens if there's more than one reference found?

File: src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ConditionMapper.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+import java.util.stream.Stream;
+
+/**
+ * Applies mapping and prefixing to condition query and conditional update expressions.
+ *
+ * @author msgroi
+ */
+class ConditionMapper {
+
+    static final String NAME_PLACEHOLDER = "#___name___";
```
Comment (msgroi) [inline src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ConditionMapper.java]: Jeremy: make it such that this name is not reserved.

**Comment by msgroi** [comment]
Reviewed in person with Jeremy.

**Pull request merged**